pkgsMusl.libffi fails in test suite

**Describe the bug**

On the latest `nixpkgs` `master` (8cc8b3b235368fad651229cfe9037cd236982f70):

```
$ nix-build -A pkgsMusl.libffi
these derivations will be built:
  /nix/store/pwcgdnmpym76ljmllzydsv7d1kvzdziw-libffi-3.3.drv
building '/nix/store/pwcgdnmpym76ljmllzydsv7d1kvzdziw-libffi-3.3.drv'...
...
running tests
check flags: SHELL=/nix/store/sydxyv59z4dka8h22if5v5p11nl1sf4w-bash-4.4-p23/bin/bash VERBOSE=y check
MAKE x86_64-pc-linux-musl : 0 * check
make[1]: Entering directory '/build/libffi-3.3/x86_64-pc-linux-musl'
Making check in include
make[2]: Entering directory '/build/libffi-3.3/x86_64-pc-linux-musl/include'
make[2]: Nothing to be done for 'check'.
make[2]: Leaving directory '/build/libffi-3.3/x86_64-pc-linux-musl/include'
Making check in testsuite
make[2]: Entering directory '/build/libffi-3.3/x86_64-pc-linux-musl/testsuite'
make  check-DEJAGNU
make[3]: Entering directory '/build/libffi-3.3/x86_64-pc-linux-musl/testsuite'
Making a new site.exp file ...
srcdir='../../testsuite'; export srcdir; \
EXPECT=expect; export EXPECT; \
if /nix/store/sydxyv59z4dka8h22if5v5p11nl1sf4w-bash-4.4-p23/bin/bash -c "runtest --version" > /dev/null 2>&1; then \
  exit_status=0; l='libffi'; for tool in $l; do \
    if runtest --tool $tool --srcdir $srcdir  ; \
    then :; else exit_status=1; fi; \
  done; \
else echo "WARNING: could not find 'runtest'" 1>&2; :;\
fi; \
exit $exit_status
WARNING: Couldn't find the global config file.
Using ../../testsuite/lib/libffi.exp as tool init file.
Test run by nixbld on Tue Feb 11 20:53:11 2020
Target is .runtest-wrapped
Host   is x86_64-pc-linux-musl

                === libffi tests ===

Schedule of variations:
    unix

Running target unix
Using /nix/store/ca489ppzi5p23dv9bk5l3dmcjsd4sasr-dejagnu-1.6.2/share/dejagnu/baseboards/unix.exp as board description file for target.
Using /nix/store/ca489ppzi5p23dv9bk5l3dmcjsd4sasr-dejagnu-1.6.2/share/dejagnu/config/unix.exp as generic interface file for target.
Using ../../testsuite/config/default.exp as tool-and-target-specific interface file.
WARNING: Assuming target board is the local machine (which is probably wrong).
You may need to set your DEJAGNU environment variable.
Running ../../testsuite/libffi.bhaible/bhaible.exp ...
FAIL: libffi.bhaible/test-call.c -W -Wall -Wno-psabi -DDGTEST=1 -Wno-unused-variable -Wno-unused-parameter -Wno-unused-but-set-variable -Wno-uninitialized -O0 (test for excess errors)
FAIL: libffi.bhaible/test-call.c -W -Wall -Wno-psabi -DDGTEST=1 -Wno-unused-variable -Wno-unused-parameter -Wno-unused-but-set-variable -Wno-uninitialized -O2 (test for excess errors)
FAIL: libffi.bhaible/test-call.c -W -Wall -Wno-psabi -DDGTEST=2 -Wno-unused-variable -Wno-unused-parameter -Wno-unused-but-set-variable -Wno-uninitialized -O0 (test for excess errors)
FAIL: libffi.bhaible/test-call.c -W -Wall -Wno-psabi -DDGTEST=2 -Wno-unused-variable -Wno-unused-parameter -Wno-unused-but-set-variable -Wno-uninitialized -O2 (test for excess errors)
...
Running ../../testsuite/libffi.call/call.exp ...
FAIL: libffi.call/align_mixed.c -W -Wall -Wno-psabi -O0 (test for excess errors)
FAIL: libffi.call/align_mixed.c -W -Wall -Wno-psabi -O2 (test for excess errors)
FAIL: libffi.call/align_stdcall.c -W -Wall -Wno-psabi -O0 (test for excess errors)
...
Running ../../testsuite/libffi.closures/closure.exp ...
Running ../../testsuite/libffi.complex/complex.exp ...
Running ../../testsuite/libffi.go/go.exp ...

                === libffi Summary ===

# of expected passes            280
# of unexpected failures        264
# of unsupported tests          206
make[3]: *** [Makefile:449: check-DEJAGNU] Error 1
make[3]: Leaving directory '/build/libffi-3.3/x86_64-pc-linux-musl/testsuite'
make[2]: *** [Makefile:526: check-am] Error 2
make[2]: Leaving directory '/build/libffi-3.3/x86_64-pc-linux-musl/testsuite'
make[1]: *** [Makefile:1347: check-recursive] Error 1
make[1]: Leaving directory '/build/libffi-3.3/x86_64-pc-linux-musl'
make: *** [Makefile:2949: check] Error 2
builder for '/nix/store/pwcgdnmpym76ljmllzydsv7d1kvzdziw-libffi-3.3.drv' failed with exit code 2
error: build of '/nix/store/pwcgdnmpym76ljmllzydsv7d1kvzdziw-libffi-3.3.drv' failed
```


**Metadata**
```
 nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 4.19.98, NixOS, 19.09pre-git (Loris)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.2`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```
Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute: libffi
```

--- a/pkgs/tools/misc/expect/default.nix
+++ b/pkgs/tools/misc/expect/default.nix
@@ -14,7 +14,7 @@ stdenv.mkDerivation rec {
 
   hardeningDisable = [ "format" ];
 
-  postPatch = ''
+  preConfigure = ''
     sed -i "s,/bin/stty,$(type -p stty),g" configure
   '';
 

